storage/google/cloud/storage/blob.py

```diff
@@ -840,52 +966,54 @@ def create_resumable_upload_session(
                                encryption#customer-supplied
 
         :type size: int
-        :param size: Optional, the maximum number of bytes that can be
-            uploaded using this session. If the size is not known when creating
-            the session, this should be left blank.
+        :param size: (Optional). The maximum number of bytes that can be
+                     uploaded using this session. If the size is not known
+                     when creating the session, this should be left blank.
 
         :type content_type: str
-        :param content_type: Optional type of content being uploaded. This can
-            be used to restrict the allowed file type that can be uploaded
-            to the size.
+        :param content_type: (Optional) Type of content being uploaded.
 
         :type origin: str
-        :param origin: Optional origin. If set, the upload can only be
-            completed by a user-agent that uploads from the given origin. This
-            can be useful when passing the session to a web client.
+        :param origin: (Optional) If set, the upload can only be completed
+                       by a user-agent that uploads from the given origin. This
+                       can be useful when passing the session to a web client.
 
-        :type client: :class:`~google.cloud.storage.client.Client` or
-                      ``NoneType``
-        :param client: Optional. The client to use.  If not passed, falls back
+        :type client: :class:`~google.cloud.storage.client.Client`
+        :param client: (Optional) The client to use.  If not passed, falls back
                        to the ``client`` stored on the blob's bucket.
 
         :rtype: str
         :returns: The resumable upload session URL. The upload can be
-            completed by making an HTTP PUT request with the file's contents.
+                  completed by making an HTTP PUT request with the
+                  file's contents.
 
         :raises: :class:`google.cloud.exceptions.GoogleCloudError`
                  if the session creation response returns an error status.
         """
-
         extra_headers = {}
-
         if origin is not None:
             # This header is specifically for client-side uploads, it
             # determines the origins allowed for CORS.
             extra_headers['Origin'] = origin
 
-        _, _, start_response = self._create_upload(
-            client,
-            size=size,
-            content_type=content_type,
-            strategy=RESUMABLE_UPLOAD,
-            extra_headers=extra_headers)
+        curr_chunk_size = self.chunk_size
+        try:
+            # Temporarily patch the chunk size. A user should still be able
```

Comment:
No action needed, but this doesn't sit easy with me. It seems we're abusing the behavior of a non-pure function in order to get it to do something it's not intended to do. It's up to you if you want to do anything about it, I won't block.